tikv-client/src/main/java/com/pingcap/tikv/types/Converter.java

```diff
@@ -88,9 +88,11 @@ public static String convertToString(Object val) {
   static byte[] convertUtf8ToBytes(Object val, int prefixLength) {
     requireNonNull(val, "val is null");
     if (val instanceof byte[]) {
-      return new String((byte[]) val).substring(0, prefixLength).getBytes(StandardCharsets.UTF_8);
+      return new String((byte[]) val).substring(0, Math.min(((byte[]) val).length, prefixLength))
+          .getBytes(StandardCharsets.UTF_8);
     } else if (val instanceof String) {
-      return ((String) val).substring(0, prefixLength).getBytes(StandardCharsets.UTF_8);
+      return ((String) val).substring(0, Math.min(((String) val).length(), prefixLength))
```

Comment:
String valStr = (String)val;
valStr.substring(0, Math.min(valStr.length(), prefixLength))
Make it clean.

Reply:
done